polynomial/core.py

```diff
@@ -446,6 +446,38 @@ def __divmod__(self, other):
 
         return Polynomial(vec), working
 
+    def __pow__(self, power, modulo=None):
+        """Return self ** power or pow(self, other, modulo)."""
+        if not isinstance(power, int):
+            raise ValueError(
+                "Can't call Polynomial() ** x with a non-integer type."
+            )
+
+        if power < 0:
+            raise ValueError(
+                "Polynomial can only be raised to a positive power."
```

Comment:
The more precise term instead of **positive** is **non-negative**, since 0 is included